scripts/postman_test_automation.sh

```diff
@@ -38,50 +45,19 @@ get_api_keys() {
 CONNECTOR_NAME="${1}"
 KEY_TYPE=""
 
-API_KEY=""
-API_SECRET=""
-KEY1=""
-
 # Function call
-get_api_keys "${CONNECTOR_NAME}"
 COLLECTION_PATH=$(path_generation "${CONNECTOR_NAME}")
+get_api_keys "${CONNECTOR_NAME}"
 
-# Run Newman collection
-case "$KEY_TYPE" in
-    "HeaderKey" )
-        args=(
-            --env-var "admin_api_key=${ADMIN_API_KEY}"
-            --env-var "baseUrl=${BASE_URL}"
-            --env-var "connector_api_key=${API_KEY}"
-        )
-        [[ -n "$GATEWAY_MERCHANT_ID" ]] && args+=("--env-var" "gateway_merchant_id=${GATEWAY_MERCHANT_ID}")
-        [[ -n "$GPAY_CERTIFICATE" ]] && args+=("--env-var" "certificate=${GPAY_CERTIFICATE}")
-        [[ -n "$GPAY_CERTIFICATE_KEYS" ]] && args+=("--env-var" "certificate_keys=${GPAY_CERTIFICATE_KEYS}")
-        newman run "${COLLECTION_PATH}" "${args[@]}"
-        ;;
-    "BodyKey" )
-        args=(
-            --env-var "admin_api_key=${ADMIN_API_KEY}"
-            --env-var "baseUrl=${BASE_URL}"
-            --env-var "connector_api_key=${API_KEY}"
-            --env-var "connector_key1=${KEY1}"
-        )
-        [[ -n "$GATEWAY_MERCHANT_ID" ]] && args+=("--env-var" "gateway_merchant_id=${GATEWAY_MERCHANT_ID}")
-        [[ -n "$GPAY_CERTIFICATE" ]] && args+=("--env-var" "certificate=${GPAY_CERTIFICATE}")
-        [[ -n "$GPAY_CERTIFICATE_KEYS" ]] && args+=("--env-var" "certificate_keys=${GPAY_CERTIFICATE_KEYS}")
-        newman run "${COLLECTION_PATH}" "${args[@]}"
-        ;;
-    "SignatureKey" )
-        args=(
-            --env-var "admin_api_key=${ADMIN_API_KEY}"
-            --env-var "baseUrl=${BASE_URL}"
-            --env-var "connector_api_key=${API_KEY}"
-            --env-var "connector_api_secret=${API_SECRET}"
-            --env-var "connector_key1=${KEY1}"
-        )
-        [[ -n "$GATEWAY_MERCHANT_ID" ]] && args+=("--env-var" "gateway_merchant_id=${GATEWAY_MERCHANT_ID}")
-        [[ -n "$GPAY_CERTIFICATE" ]] && args+=("--env-var" "certificate=${GPAY_CERTIFICATE}")
-        [[ -n "$GPAY_CERTIFICATE_KEYS" ]] && args+=("--env-var" "certificate_keys=${GPAY_CERTIFICATE_KEYS}")
-        newman run "${COLLECTION_PATH}" "${args[@]}"
-        ;;
-esac
+# Newman runner
+newman run "${COLLECTION_PATH}" \
+    --env-var "admin_api_key=${ADMIN_API_KEY}" \
+    --env-var "baseUrl=${BASE_URL}" \
+    --env-var "connector_api_key=${API_KEY}" \
```

Comment:
Instead of `export`ing these as environment variables, can you use single quotes around these values? Would that help prevent the shell from reading values with spaces as separate arguments?

```suggestion
    --env-var "admin_api_key='${ADMIN_API_KEY}'" \
    --env-var "baseUrl='${BASE_URL}'" \
    --env-var "connector_api_key='${API_KEY}'" \
```

Same change for the following lines?

Also, just being sure, would these API keys be printed in the logs, either when the script is run or by `newman`?

Reply:
This cannot be done. Using single quotes directly considers the API_KEYS along with single quotes like `'sdkjfdskdgd'` instead of `sdkjfdskdgd`. 
The same behaviour can be observed when trying to print the contents of a variable as well. If you do `echo '${something}'`, it will print `${something}` instead of the contents within it. However, when using it as a command, it prints the contents along with single quotes.  
Unfortunately, when you run Payment Connector - Create, yes, API keys do get printed. Thanks for reminding that, I feel like it is better to remove logs all over for safety and use it whenever needed locally.

Reply:
> This cannot be done. Using single quotes directly considers the API_KEYS along with single quotes like `'sdkjfdskdgd'` instead of `sdkjfdskdgd`.

> The same behaviour can be observed when trying to print the contents of a variable as well. If you do `echo '${something}'`, it will print `${something}` instead of the contents within it. However, when using it as a command, it prints the contents along with single quotes.

These two are independent things: the first one is about how `newman` parses environment variables, the second is common shell behavior.

Can you do without `export`ing them? As far as I understand, since you aren't using `local`, the variables should be accessible anywhere within the script, but not outside it. Something like this: `MY_VAR=value`?